RecoTracker/MkFit/python/customizeHLTIter0ToMkFit.py

```diff
@@ -11,76 +11,81 @@
 import RecoLocalTracker.SiStripRecHitConverter.SiStripRecHitConverter_cfi as SiStripRecHitConverter_cfi
 
 def customizeHLTIter0ToMkFit(process):
-    # mkFit needs all clusters, so switch off the on-demand mode
-    process.hltSiStripRawToClustersFacility.onDemand = False
 
-    process.hltSiStripRecHits = SiStripRecHitConverter_cfi.siStripMatchedRecHits.clone(
-        ClusterProducer = "hltSiStripRawToClustersFacility",
-        StripCPE = "hltESPStripCPEfromTrackAngle:hltESPStripCPEfromTrackAngle",
-        doMatching = False,
-    )
+    try:
```

Comment:
What's the expected behavior here for the new HLT setup, is it no customization?
Offhand, which part fails? (the proposed change is a bit too inclusive "if anything fails, skip")

Reply:
> What's the expected behavior here for the new HLT setup, is it no customization?

Yes, no customisation. In a "fake" HLT menu (2022 wfs), there is no tracking to customise.

> Offhand, which part fails?

[The very first line](https://github.com/cms-sw/cmssw/blob/bfd82ebacf422e9cb7de549173217580a26b59c8/RecoTracker/MkFit/python/customizeHLTIter0ToMkFit.py#L15). `hltSiStripRawToClustersFacility`, `HLTDoLocalStripSequence`, `HLTIterativeTrackingIteration0` and `hltIter0PFlowCkfTrackCandidates` are not present in "fake" HLT menus.

> (the proposed change is a bit too inclusive "if anything fails, skip")

I agree, but it's intentional. I would like to stop this customisation from being an issue for HLT (having in mind https://github.com/cms-sw/cmssw/pull/34735#issuecomment-893219825, and previous occurrences like https://github.com/cms-sw/cmssw/pull/38816#issuecomment-1191746445). Alternatively, I can revert the changes and follow [what was agreed](https://github.com/cms-sw/cmssw/pull/33802#issuecomment-861679963) (let `*.7` wfs fail, and expect ORP to merge HLT PRs regardless), but to me that's not a real solution. (also, the premise of https://github.com/cms-sw/cmssw/pull/33802#issuecomment-861753979 slightly changed, as a `*.7` wf is now part of the PR tests)

Just for my understanding, is it really useful to apply this HLT customisation to step2 of wfs `*.7` ?

Reply:
> [The very first line](https://github.com/cms-sw/cmssw/blob/bfd82ebacf422e9cb7de549173217580a26b59c8/RecoTracker/MkFit/python/customizeHLTIter0ToMkFit.py#L15). `hltSiStripRawToClustersFacility`, `HLTDoLocalStripSequence`, `HLTIterativeTrackingIteration0` and `hltIter0PFlowCkfTrackCandidates` are not present in "fake" HLT menus.

I think that it's more practical to check for existence of e.g. `hltSiStripRawToClustersFacility` and then return.
The current solution would mask all config problems in a valid HLT configuration.

Reply:
>  The current solution would mask all config problems in a valid HLT configuration.

I see your point, but "practical" is subjective.

What is practical for HLT is that this customisation does not break anymore when HLT makes its own PRs. If we give the function the opportunity to break, I expect it will break again in the future.

As I wrote, HLT can leave it broken, and explain everytime that PRs are expected to be merged regardless, but I was hoping to find a better solution.

Maybe one option is to remove wfs `*.7` from PR tests to return to the premise of https://github.com/cms-sw/cmssw/pull/33802#issuecomment-861753979, then the relevant experts can update the customisation once IB failures appear.

Reply:
Could 5bd05eb2c15 be considered an improvement?

> Just for my understanding, is it really useful to apply this HLT customisation to step2 of wfs *.7 ?

I would still be (genuinely) curious to know how this HLT customisation is useful.

Reply:
> Could [5bd05eb](https://github.com/cms-sw/cmssw/commit/5bd05eb2c15768841b6e173d06f682d41eb6f371) be considered an improvement?

Yes, thank you.

> 
> > Just for my understanding, is it really useful to apply this HLT customisation to step2 of wfs *.7 ?
> 
> I would still be (genuinely) curious to know how this HLT customisation is useful.

There is still a plan to use mkFit for track building in HLT; one of the components towards that was #34618 which was merged recently.

Reply:
(5bd05eb was slightly improved in abe766f5cfc.)